source/ui/menus/createMenu.d

```diff
@@ -0,0 +1,446 @@
+import core.sudoku.sudoku;
+import core.sudokuType;
+import ui.sudoku.gridUI;
+
+class CreateMenu
+{
+	public this(Builder builder)
+	{
+		trace("Loading CreateMenu widgets...");
+		btnBoxRules = cast(ButtonBox) builder.getObject("btnBoxRules");
+		btnSave = cast(Button) builder.getObject("btnSave");
+		cbSudokuType = cast(ComboBox) builder.getObject("cbSudokuType");
+		ckBtnCreateSolution = cast(CheckButton) builder.getObject("ckBtnCreateSolution");
+		ckBtnCustomSudoku = cast(CheckButton) builder.getObject("ckBtnCustomSudoku");
+		lblColumn = cast(Label) builder.getObject("lblColumn");
+		lblDigit = cast(Label) builder.getObject("lblDigit");
+		lblRow = cast(Label) builder.getObject("lblRow");
+		lblSwitchSolution = cast(Label) builder.getObject("lblSwitchSolution");
+		stkMiddle = cast(Stack) builder.getObject("stackChoiceMiddle");
+		stkRules = cast(Stack) builder.getObject("stackRules");
+		stkSwitcherRules = cast(StackSwitcher) builder.getObject("stkSwitcherRules");
+		switchSolution = cast(Switch) builder.getObject("switchSolution");
+		txtDescription = cast(Entry) builder.getObject("txtDescription");
+		txtSudokuTitle = cast(Entry) builder.getObject("txtSudokuTitle");
+		txtUsername = cast(Entry) builder.getObject("txtUsername");
+		initCbSudokuType();
+		initBtnBoxRules();
+	}
+
+
+	// update cbSudokuType values
+	private void initCbSudokuType()
+	{
+		auto list = new ListStore([GType.STRING]);
+		cbSudokuType.setModel(list);
+		foreach (i, type; EnumMembers!SudokuType)
+		{
+			auto iter = list.createIter();
+			list.setValue(iter, 0, type);
+			static if (i == 0)
+			{
+				cbSudokuType.setActiveIter(iter);
+			}
+		}
+	}
+
+
+	// update btnBoxRules values
+	private void initBtnBoxRules()
+	{
+		foreach (type; EnumMembers!RuleType)
+		{
+			auto ckBtn = new CheckButton(type);
+			btnBoxRules.packStart(ckBtn, true, true, 0);
+		}
+		btnBoxRules.showAll();
+	}
+
+
+
+// signals
+
+	public void setOnBtnSaveClicked(void delegate(Button) dg)
+	{
+		btnSave.addOnClicked(dg);
+	}
+
+
+	public void setOnCbSudokuTypeChanged(void delegate(ComboBox) dg)
+	{
+		cbSudokuType.addOnChanged(dg);
+	}
+
+
+	public void setOnCkBtnCreateSolutionToggled(void delegate(ToggleButton) dg)
+	{
+		ckBtnCreateSolution.addOnToggled(dg);
+	}
+
+
+	public void setOnCkBtnCustomSudokuToggled(void delegate(ToggleButton) dg)
+	{
+		ckBtnCustomSudoku.addOnToggled(dg);
+	}
+
+
+	public void setOnRuleToggled(void delegate(ToggleButton) dg)
+	{
+		foreach (child; getBtnBoxRulesChildren())
+		{
+			child.addOnToggled(dg);
+		}
+	}
+
+
+	public void setOnSwitchSolutionstateSet(bool delegate(bool, Switch) dg)
+	{
+		switchSolution.addOnStateSet(dg);
+	}
+
+
+
+// functions
+
+	public void addGrid(GameState state, GridUI gridUI)
+	{
+		stkMiddle.addNamed(gridUI, state);
+	}
+
+
+	public void hideSwitchSolution()
+	{
+		switchSolution.setSensitive(false);
+		switchSolution.setVisible(false);
+		switchSolution.setState(false);
+		switchSolution.setActive(false);
```

Comment:
Pretty sure just a `setActive` or `setVisible` is enough

Reply:
For what I need, `setVisible` is enough, this were experiments I did in the beginning and then forgot about them.